File: BotsSDK/widgetsdk/src/main/java/com/kore/ai/widgetsdk/utils/NetworkUtility.java
package com.kore.ai.widgetsdk.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtility {

    public final static int HTTP_ERROR_CODE_403 = 403;
    public final static String HTTP_SCHEME = "http";
    public final static String FILE_SCHEME = "file:";
    public final static String CONTENT_SCHEME = "content:";

    /**
     * Method to check Network Connections
     *
     * @param context
     * @return boolean value
     */

    public static boolean isNetworkConnectionAvailable(Context context) {
        return isNetworkConnectionAvailable(context, true);
    }

    public static boolean isNetworkConnectionAvailable(Context context, boolean showMessage) {
        boolean isNetworkConnectionAvailable = false;

        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();

        if (activeNetworkInfo != null) {
            isNetworkConnectionAvailable = activeNetworkInfo.isConnected();
        }

        if (!isNetworkConnectionAvailable && showMessage) {
//            CustomToast.showToast(context, "No network available");

        }
        return isNetworkConnectionAvailable;
    }
}

File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/common/utils/NetworkUtility.kt
package kore.botssdk.common.utils

import android.content.Context
import android.net.ConnectivityManager

class NetworkUtility {

    companion object {
        const val HTTP_ERROR_CODE_403 = 403
        const val HTTP_SCHEME = "http"
        const val FILE_SCHEME = "file:"
        const val CONTENT_SCHEME = "content:"

        /**
         * Method to check Network Connections
         *
         * @param context
         * @return boolean value
         */
        fun isNetworkConnectionAvailable(context: Context): Boolean {
            return isNetworkConnectionAvailable(context, true)
        }

        private fun isNetworkConnectionAvailable(context: Context, showMessage: Boolean): Boolean {
            var isNetworkConnectionAvailable = false
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetworkInfo = connectivityManager.activeNetworkInfo
            if (activeNetworkInfo != null) {
                isNetworkConnectionAvailable = activeNetworkInfo.isConnected
            }
            if (!isNetworkConnectionAvailable && showMessage) {
//            CustomToast.showToast(context, "No network available");
            }
            return isNetworkConnectionAvailable
        }
    }
}